modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/ItemActions.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDropDown, {Align} from '@clayui/drop-down';
+import ClayIcon from '@clayui/icon';
+import {useModal} from '@clayui/modal';
+import {useIsMounted} from 'frontend-js-react-web';
+import PropTypes from 'prop-types';
+import React, {useEffect, useMemo, useState} from 'react';
+
+import {getLayoutDataItemPropTypes} from '../../prop-types';
+import {
+	BACKSPACE_KEYCODE,
+	D_KEYCODE,
+	S_KEYCODE,
+} from '../config/constants/keycodes';
+import {LAYOUT_DATA_ITEM_TYPES} from '../config/constants/layoutDataItemTypes';
+import selectCanUpdatePageStructure from '../selectors/selectCanUpdatePageStructure';
+import {useDispatch, useSelector} from '../store/index';
+import deleteItem from '../thunks/deleteItem';
+import duplicateItem from '../thunks/duplicateItem';
+import {useIsActive, useSelectItem} from './Controls';
+import SaveFragmentCompositionModal from './floating-toolbar/SaveFragmentCompositionModal';
+import hasDropZoneChild from './layout-data-items/hasDropZoneChild';
+
+export default function ItemActions({item}) {
+	const [active, setActive] = useState(false);
+	const dispatch = useDispatch();
+	const isActive = useIsActive();
+	const isMounted = useIsMounted();
+	const selectItem = useSelectItem();
+
+	const canUpdatePageStructure = useSelector(selectCanUpdatePageStructure);
+	const state = useSelector((state) => state);
+	const {
+		fragmentEntryLinks,
+		layoutData,
+		segmentsExperienceId,
+		widgets,
+	} = state;
+
+	const [
+		openSaveFragmentCompositionModal,
+		setOpenSaveFragmentCompositionModal,
+	] = useState(false);
+
+	const {observer, onClose} = useModal({
+		onClose: () => {
+			if (isMounted()) {
+				setOpenSaveFragmentCompositionModal(false);
+			}
+		},
+	});
+
+	useEffect(() => {
+		const onKeyDown = (event) => {
+			if (isActive(item.itemId)) {
+				const itemAction = itemActions.find((itemAction) =>
+					itemAction.isKeyCombination(event)
+				);
+
+				if (itemAction) {
+					itemAction.action();
+
+					event.preventDefault();
+				}
+			}
+		};
+
+		document.addEventListener('keydown', onKeyDown, true, window);
```

Comment:
I think it should be window.addEventListener, right? 🤔  [addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) doesn't seem to receive a 4th argument

Reply:
You are right, thanks! Modified 🙂